.github/workflows/ci.yml

```diff
@@ -24,51 +24,38 @@ jobs:
             experimental: true
 
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@9cd00a88a73addc8617065438eff914dd08d0955 # 2023-02-02
```

Comment:
Why use this action? And why is it at a git hash not a version?

Reply:
The author has expressed they dont want to provide tags except for when its used to specify rust versions. Afaik, `uses` can not be evaluated like `uses: ${{ matrix.var }}`

Reply:
checkout@v4